This patch has been merged upstream and shouldn't be needed for 1.9.1:
https://github.com/libretro/RetroArch/pull/11170

diff --git a/libretro-common/file/nbio/nbio_linux.c b/libretro-common/file/nbio/nbio_linux.c
index 70e3a98..0c2d857 100644
--- libretro-common/file/nbio/nbio_linux.c
+++ libretro-common/file/nbio/nbio_linux.c
@@ -36,6 +36,7 @@
 #include <fcntl.h>
 #include <sys/syscall.h>
 #include <linux/aio_abi.h>
+#include <time.h>
 
 struct nbio_linux_t
 {
